### ISSUE TITLE
[HOLD for payment 2023-06-12] [$1000] Remove "To" label for participant selector in case of split bill and send/request money.

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#19233]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Click on the Fab button in the LHN
2. Click on the split bill button.
3. Enter the amount and click next.
4. Notice that before the text input, there is a label called "To" at the top.
5. Close this modal.
6. Go back to LHN and click on the Fab again
7. Click on either send money or request money button.
8. Enter the amount and click next.
9. Notice that before the text input, there is no label like "To".

## Expected Result: 
Remove the `To` label from all areas of the UI. The behavior should be consistent. 

## Actual Result: 
 The `To` label is shown in participant selection component in case of "Split bill" but not in case of "Send/Request money".

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.16-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/d74f6563-6676-404b-9baa-e76c9604f4da

https://github.com/Expensify/App/assets/43996225/8e7a916f-3860-453e-bbe9-2f84168c4f3e

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Nikhil-Vats
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684262848278309

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~010aadb10658e3788e</li>
        <li>Upwork Job ID: 1661137551238852608</li>
        <li>Last Price Increase: 2023-05-30</li>
    </ul>
</details>

**Comment by kavimuru** [comment]
Proposal by @Nikhil-Vats
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
Inconsistent label for participant selector in case of split bill and send/request money.
### What is the root cause of that problem?
In `MoneyRequestParticipantsPage`, if the prop `hasMultipleParticipants` is true then we render `MoneyRequestParticipantsSplitSelector` (Split bill) otherwise we render `MoneyRequestParticipantsSelector` (Send/Request money).
The `MoneyRequestParticipantsSplitSelector` component uses `Text` for showing the label "To" before `OptionSelector` -
https://github.com/Expensify/App/blob/05f5b06e37dafdeff5a86441c615557a6e735c3c/src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsSplitSelector.js#L215-L218
However, there is no use of `Text` in `MoneyRequestParticipantsSelector`, it directly returns `OptionSelector` -
https://github.com/Expensify/App/blob/05f5b06e37dafdeff5a86441c615557a6e735c3c/src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsSelector.js#L156-L159
### What changes do you think we should make in order to solve the problem?
We should use `View` as a wrapper in `MoneyRequestParticipantsSelector` which is shown in case of Send/Request money and use `Text` to show the correct label for this case. This will make the behaviour consistent across all IOU request types (split, send, request).
### What alternative solutions did you explore? (Optional)
NA

**Comment by Nikhil-Vats** [comment]
@kavimuru Thanks for posting my proposal. I would like to point out that this is a discrepancy on other pages/modals also -

**Without label** -
1. NewChatPage (Click on FAB then click on New Chat)
2. SearchPage (Click on the search icon in the header in LHN)
3. YearPickerPage (Go to settings, then your profile, click on personal details, Date of birth and then year)
4. IOUCurrencySelection (Change currency in any type of IOU request)
5. **MoneyRequestParticipantsSelector** (The one this issue is regarding - Send/Request money)
6. TimezoneSelectPage (Settings > Profile > Timezone)
7. TaskAssigneeSelectorModal (Click on FAB then assign task, enter a title and click next and click assignee)
8. TaskShareDestinationSelectorModal (Click on FAB then assign task, enter a title and click next and click share somewhere)
9. WorkspaceInvitePage (Settings > Workspaces > Any workspace > Manage members > Invite)

**With label** - 
1. **MoneyRequestParticipantsSplitSelector** (The one mentioned in the issue - Split bill) - Shows a label "To".
2. PronounsPage (Go to settings then profile) - It shows not a label but a description - "Your pronouns are shown on your profile.".

Now, some of the components without label don't need a label such as SearchPage, YearPickerPage, IOUCurrencySelection, etc. but between similar pages like Request Money, Send Money, Split bill, etc. and also the Task assignee, share destination modals it would be nice to have consistency.

We can have labels for these components as I have mentioned in the solution section of my proposal or we can remove labels to keep consistency between similar components.

**Comment by michaelhaxhiu** [comment]
I agree this is inconsistent and worth fixing.

**Comment by michaelhaxhiu** [comment]
Good question, I am not sure yet. I want to figure out when we added the `To` field and the intentions behind adding it. Then it'll be easier to decide this more clearly!

**Comment by michaelhaxhiu** [comment]
We need to find a consensus on the thread in slack. Then we can re-apply `Help Wanted` and proceed. I could see us removing the `To` field but I'm just 1 vote here. There's a chance that we added it for good reason and keep it, perhaps even adding it to all the UI components where you select a person (e.g. from the FAB flows)

**Comment by aldo-expensify** [comment]
No updates, waiting on Shawn to come back to get his input

**Comment by michaelhaxhiu** [comment]
We discussed further and @shawnborton weighed in. We agree that it's wise to kill the `To:` field in all areas of the UI.

**Comment by michaelhaxhiu** [comment]
Updated the GH body, expected result, and title to be inline with our expectations.

@Nikhil-Vats since you already kicked off a proposal, just wanna nudge you in case you need to adjust/adapt that proposal to fit the latest agreed-upon plan.

**Comment by Nikhil-Vats** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Inconsistent "To" label for participant selector in case of split bill and send/request money. The label "To" in split bill participant selector is unnecessary since the UI is self explanatory. 

### What is the root cause of that problem?
In `MoneyRequestParticipantsPage`, if the prop `hasMultipleParticipants` is true then we render `MoneyRequestParticipantsSplitSelector` (Split bill) otherwise we render `MoneyRequestParticipantsSelector` (Send/Request money).

The `MoneyRequestParticipantsSplitSelector` component uses `Text` for showing the label "To" before `OptionSelector` -
https://github.com/Expensify/App/blob/05f5b06e37dafdeff5a86441c615557a6e735c3c/src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsSplitSelector.js#L215-L218

However, these is no use of `Text` in `MoneyRequestParticipantsSelector`, it directly returns `OptionSelector` -
https://github.com/Expensify/App/blob/05f5b06e37dafdeff5a86441c615557a6e735c3c/src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsSelector.js#L156-L159

### What changes do you think we should make in order to solve the problem?
We should remove line 217 (see below) from `MoneyRequestParticipantsSplitSelector` to remove the `To` label since the UI is self explanatory.

```javascript
<Text style={[styles.textLabelSupporting, styles.pt3, styles.ph5]}>{this.props.translate('common.to')}</Text> 
```
We cannot remove the view wrapper because it is used for safe area padding (near the next button).
<details>
<summary>
See screenshot.
</summary>

<img width="375" alt="image" src="https://github.com/Expensify/App/assets/31413407/f7bc941b-baa1-45d4-a572-ce6e86a800a0">

</details>
### What alternative solutions did you explore? (Optional)

NA

**Comment by aldo-expensify** [comment]
Thanks @aimane-chnaif

**Comment by Nikhil-Vats** [comment]
Thanks for assigning me this issue. I have applied to the job on Upwork and I will raise the PR within a few hours.

**Comment by Nikhil-Vats** [comment]
Hey @aimane-chnaif @michaelhaxhiu I have raised the PR. You can review and test it. Some of the platform videos (test proofs) are pending, I will add them by EOD today.

**Comment by Nikhil-Vats** [comment]
Hey all 👋, I have added the videos for all platforms now. All wrapped up from my end. You can review and test.

**Comment by michaelhaxhiu** [comment]
<img width="394" alt="image" src="https://github.com/Expensify/App/assets/13206896/cfb42717-7407-4182-9ec0-f999fef2a234">

**Comment by michaelhaxhiu** [comment]
Invited yall to the job. We'll sort out final payout amounts after 7 days passes. 

So far though, the 50% bonus is obviously in play

**Comment by michaelhaxhiu** [comment]
I agree w/ that take

**Comment by michaelhaxhiu** [comment]
all paid. thanks to all! 😎 till next time, folks.
